cpp/models/ide_secir/model.cpp

```diff
@@ -242,213 +315,283 @@ void Model::compute_flow(Eigen::Index idx_InfectionTransitions, Eigen::Index idx
+        Eigen::Index calc_time_index = (Eigen::Index)std::ceil(calc_time / dt) - 1;
+
+        Eigen::Index num_time_points;
+        ScalarType current_time;
+
+        if (initialization) {
+            // Determine m_forceofinfection at time t0-dt which is the penultimate timepoint in m_transitions.
+            num_time_points = m_transitions.get_num_time_points() - 1;
+            // Get time of penultimate timepoint in m_transitions.
+            current_time = m_transitions.get_time(num_time_points - 1);
+        }
+        else {
+            // Determine m_forceofinfection for current last time in m_transitions.
+            num_time_points = m_transitions.get_num_time_points();
+            current_time    = m_transitions.get_last_time();
+        }
+        //We compute the Season Value.
         ScalarType season_val =
             1 +
             parameters.get<Seasonality>() *
                 sin(3.141592653589793 * (std::fmod((parameters.get<StartDay>() + current_time), 365.0) / 182.5 + 0.5));
-        m_forceofinfection +=
-            season_val * parameters.get<TransmissionProbabilityOnContact>().eval(state_age) *
-            parameters.get<ContactPatterns>().get_cont_freq_mat().get_matrix_at(current_time)(0, 0) *
-            ((parameters
-                      .get<TransitionProbabilities>()[(int)InfectionTransition::InfectedNoSymptomsToInfectedSymptoms] *
-                  parameters
-                      .get<TransitionDistributions>()[(int)InfectionTransition::InfectedNoSymptomsToInfectedSymptoms]
-                      .eval(state_age) +
-              parameters.get<TransitionProbabilities>()[(int)InfectionTransition::InfectedNoSymptomsToRecovered] *
-                  parameters.get<TransitionDistributions>()[(int)InfectionTransition::InfectedNoSymptomsToRecovered]
-                      .eval(state_age)) *
-                 m_transitions[i + 1][Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)] *
-                 parameters.get<RelativeTransmissionNoSymptoms>().eval(state_age) +
-             (parameters.get<TransitionProbabilities>()[(int)InfectionTransition::InfectedSymptomsToInfectedSevere] *
-                  parameters.get<TransitionDistributions>()[(int)InfectionTransition::InfectedSymptomsToInfectedSevere]
-                      .eval(state_age) +
-              parameters.get<TransitionProbabilities>()[(int)InfectionTransition::InfectedSymptomsToRecovered] *
-                  parameters.get<TransitionDistributions>()[(int)InfectionTransition::InfectedSymptomsToRecovered].eval(
-                      state_age)) *
-                 m_transitions[i + 1][Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)] *
-                 parameters.get<RiskOfInfectionFromSymptomatic>().eval(state_age));
+        // We need to sum, over contacts with all Age Groups.
+        for (auto j = AgeGroup(0); j < AgeGroup(m_num_agegroups); ++j) {
+
+            ScalarType deaths_j;
```

Comment:
Can you move this directly before if statement below and add a comment what will be stored there?